core/src/test/java/org/elasticsearch/indices/stats/IndexStatsIT.java

```diff
@@ -462,6 +465,33 @@ public void simpleStats() throws Exception {
         assertThat(stats.getTotal().getIndexing(), nullValue());
         assertThat(stats.getTotal().getGet(), nullValue());
         assertThat(stats.getTotal().getSearch(), nullValue());
+
+        // index failed
+        try {
+            client().prepareIndex("test1", "type1", Integer.toString(1)).setSource("field", "value").setVersion(1)
+                    .setVersionType(VersionType.EXTERNAL).execute().actionGet();
+        } catch (VersionConflictEngineException e) {
+            //expected
+        }
+        try {
+            client().prepareIndex("test1", "type2", Integer.toString(1)).setSource("field", "value").setVersion(1)
+                    .setVersionType(VersionType.EXTERNAL).execute().actionGet();
+        } catch (VersionConflictEngineException e) {
+            //expected
+        }
+        try {
+            client().prepareIndex("test2", "type", Integer.toString(1)).setSource("field", "value").setVersion(1)
+                    .setVersionType(VersionType.EXTERNAL).execute().actionGet();
+        } catch (VersionConflictEngineException e) {
+            //expected
+        }
+        refresh();
+
+        stats = client().admin().indices().prepareStats().setTypes("type1").execute().actionGet();
```

Comment:
@nik9000 actually I only need it because of the assertion: 
```java
assertThat(stats.getPrimaries().getIndexing().getTypeStats().get("type1").getIndexFailedCount(), equalTo(1L));
```
I dont think we need to check for each type, but I can add it if you think its needed.